### DISCUSSION TITLE
The configuration nuxt.config.ts file is an error reported by the TS compiler

### DISCUSSION THREAD

**Discussion by Firstrings** [nuxt/nuxt#21123] [Questions]
When I tried to configure the corresponding attribute in nuxt.jsconfig, ts reported an error stating that this attribute does not exist, but nuxt can be compiled.The details are as follows:

```
//  nuxt.config.ts
export default defineNuxtConfig({
    runtimeConfig: {       // error-detail -> Parameters of type '{runtimeConfig: {apiSecret: number; public: {apiBase: string;};};}' cannot be 
                                         assigned to parameters of type 'InputConfig<NuxtConfig, ConfigLayerMeta>'.
                                        Object text can only specify known attributes, and 'runtimeConfig' is not in the type 
                                        'InputConfig<NuxtConfig, ConfigLayerMeta>'
        apiSecret: 123,
        public: {
            apiBase: '/api'
        }
    }
})
```
```
//  package.json
{
  "name": "nuxt-app",
  "private": true,
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "devDependencies": {
    "@types/node": "^18",
    "nuxt": "^3.5.1"
  }
}
```
```
// nuxt.config.ts
If I use the following method, I won't report an error：export default {
  app: {
    head: {
      charset: 'utf-8',
      title: 'My App',
      htmlAttrs: {
        lang: "en"
      },
      meta: [
        { name: 'description', content: 'My amazing site.' }
      ],
    }
  },
  css: ['@fortawesome/fontawesome-svg-core/styles.css']
}
```
May I ask what is the reason for this? Thank you very much for your help！

**Comment by warflash** [comment]
Are you using webstorm by chance?

**Reply by Firstrings** [reply]
Thank you for your reply. I am currently using vscode

**Reply by notflip** [reply]
I am using Webstorm and I'm getting the error `... and 'app' does not exist in type 'InputConfig<NuxtConfig, ConfigLayerMeta`
Is this a webstorm issue?

**Reply by warflash** [reply]
@notflip try upgrading your webstorm ts version to >5

**Reply by notflip** [reply]
@warflash That fixes it! Thanks a lot. For anyone who's wondering how to update the Webstorm TS version, just include typescript 5 in the project's package.json, Webstorm will use that one first.

**Reply by gspgsp** [reply]
I also have this problem (v3.5.2), when I configure the build parameters, I get the following error: 'build' does not exist in type 'inputconfig<nuxtconfig, configlayermeta>'... , but the project works. If it's the ts version you're talking about, I think I'll try upgrading.

**Reply by notflip** [reply]
@gspgsp This will be fixed using TS > 5, I've also had the build item being shown as erronous!

**Comment by Kokleng-Dev** [comment]
@warflash how about vscode? 
i got this error too.

<img width="1436" alt="Screenshot 2024-10-12 at 1 56 08 in the afternoon" src="https://github.com/user-attachments/assets/89e1b530-005b-40b1-9ad4-833ae7a5f0d7">
